### PR TITLE
add lsif types

### LINKED ISSUE #214: Adding LSIF types
LSIF is a JSON format for dumping knowledge of a language server about a code base. It can be used by code viewers to provide ide features without running a language server online.

There is a [dead crate](https://github.com/Xanewok/lsif) that provides LSIF types, and with [minimal changes](https://github.com/hkalbasi/rust-analyzer/blob/master/crates/rust-analyzer/src/cli/lsif/lsif_types.rs) it can be used today. It can live in a seperate crate, but since it depends on `lsp_types` and I think every user of it also depend on `lsp_types` too, for preventing conflicts and for making `lsp_types` and `lsif_types` considering each other in changes, I think it is good to have lsif types inside this repository.

I can file a PR if it is ok to have lsif types inside this repository.

cc https://github.com/rust-analyzer/rust-analyzer/pull/10181

### PR DESCRIPTION
**Pull request by HKalbasi**
Fix #214

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -104,6 +104,8 @@ pub use workspace_folders::*;
 mod workspace_symbols;
 pub use workspace_symbols::*;
 
+pub mod lsif;
```
Comment (Marwes) [inline src/lib.rs:107]: Could you add a comment explaining `lsif`/linking to the definition for it here so it shows up in docs?

**Comment by Marwes** [comment]
Thanks!

**Comment by HKalbasi** [comment]
@Marwes Can you please make a release in crates.io? So we can use it in rust analyzer.

**Comment by HKalbasi** [comment]
@Marwes A friendly reminder. The rust analyzer lsif PR is blocked on releasing this to crates.io. If there is a problem for releasing or there is a timing / plan for releases, please tell me not to bother you with notification.

**Comment by Marwes** [comment]
Thanks for the ping, I just forgot (busy week). Released as 0.90

**Comment by krobelus** [comment]
If you're ever blocked again, you can temporarily depend on an unreleased version, see https://stackoverflow.com/questions/54196846/how-to-specify-a-certain-commit-in-dependencies-in-cargo-toml

**Pull request merged**